CONTRIBUTING.md

```diff
@@ -1,92 +1,146 @@
    it's great to know that you have a clean slate.
 
-3. Add a test for your change. Only refactoring and documentation
-   changes require no new tests. If you are adding functionality
+1. Add a test for your change.  Only refactoring and documentation
+   changes require no new tests.  If you are adding functionality
    or fixing a bug, please add a test.
 
-4. Make the test pass.
+1. Make the test pass.
+
+1. Push your fork and submit a pull request.
+
+## Ruby dependencies
+
+This project is managed with
+[PDK](https://puppet.com/try-puppet/puppet-development-kit/) and based on the
+[PDK template](https://github.com/puppetlabs/pdk-templates).
+
+Many files in this repo are generated by PDK based on the configuration in the
+`.sync.yml` file. Changes to files generated by PDK should only go through
+changes to the `.sync.yml` and then be updated via `pdk update`.
+
+For your convenience the `pdk` is also installed by the `Gemfile` in this repo
+and [bundler](http://bundler.io/) and once the bundler project is properly
+configured and installed can be called via `bundler exec pdk`.
+
+Bundler manages the development tools and library ruby dependencies via the
+`Gemfile`, which is generated by the PDK in this project.
+
+PDK provides a couple of environment variables to modify the versions of
+[puppet](https://puppet.com/docs/puppet),
+[facter](https://puppet.com/docs/puppet/latest/facter.html) and
+[hiera](https://puppet.com/docs/puppet/latest/hiera.html), which can be set
+before fetching all dependencies.
+
+For example:
+
+```bash
+export PUPPET_GEM_VERSION="~> 7.17.0"
+export FACTER_GEM_VERSION="~> 2.5.7"
+export HIERA_GEM_VERSION="~> 3.9.0"
+```
+
+You might want to specify a project local directory to fetch all dependencies
+into before installing all of them:
+
+```bash
+bundle config set --local path vendor/bundle
+```
+
+Installing the ruby dependencies is done by:
 
-5. Push to your fork and submit a pull request.
+```bash
+bundle install
+```
 
+## PDK, Syntax and style
 
-## Dependencies
+Afterward, you have access to the PDK and all other tools like this:
 
-The testing and development tools have a bunch of dependencies,
-all managed by [Bundler](http://bundler.io/) according to the
-[Puppet support matrix](http://docs.puppetlabs.com/guides/platforms.html#ruby-versions).
+```bash
+bundle exec pdk
+```
 
-By default the tests use a baseline version of Puppet.
+Also static and style checks are now possible.  For instance checking the puppet
+and ruby code can be done like this:
 
-If you have Ruby 2.x or want a specific version of Puppet,
-you must set an environment variable such as:
+```bash
+bundle exec rake markdownlint validate lint check rubocop
+```
 
-    export PUPPET_VERSION="~> 3.2.0"
+Checking the Markdown files with
+[markdownlint](https://github.com/markdownlint/markdownlint) can be done like
+this:
 
-Install the dependencies like so...
+```bash
+bundle exec rake markdownlint
```

Comment:
That is strange, because the `markdownlint` task is implemented in `Rakefile.local` and that is 'included' in the `Rakefile` via the `require Rakefile.local` line. So if `rake` finds the right `Rakefile` it should also know about that tasks...

Reply:
There might be some local configuration issue. I think we can just leave it like this.